### PR TITLE
Lps 182296 followup

### PR DESCRIPTION
**Pull request by slnn**
Resend according to the comment in  https://github.com/liferay-core-infra/liferay-portal/pull/4381 

Hi @yuhai 

I have double checked these:

1. Portal will trigger reindex and insert data into AssertEntry when create a DL Folder from UI, so when modify the name of the DL Folder, portal will retrigger reindex and update the data in AssertEntry
2. Portal will trigger reindex but not insert data into AssertEntry when portal creates these temporary dl folders which this pull related

So I just call this method in DLFolderLocalService to update the name in DLFolder, I do not call any APIs to update AssetEntry since I do not see any data are inserted.

```
@Indexable(type = IndexableType.REINDEX)
	public DLFolder updateDLFolder(DLFolder dlFolder);
```

/cc @dantewang @tinatian 

Thanks
Lily

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:test:relevant

File: modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/upgrade/v1_0_1/DLFolderNameUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			}
+		}
+
+		DLFolder dlFolder = _dlFolderLocalService.getDLFolder(folderId);
+
+		dlFolder.setName(UploadImageUtil.TEMP_IMAGE_FOLDER_NAME);
+
+		_dlFolderLocalService.updateDLFolder(dlFolder);
+	}
+
```
Comment (yuhai) [inline modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/upgrade/v1_0_1/DLFolderNameUpgradeProcess.java:58]: @slnn, I find there are two issues in the fix.

1. From code logic seen, you should move the blow codes to while loop.  You may directly use `_dlFolderLocalService.getDLFolder(resultSet.getLong("folderId"))` to get the Folder. This also will avoid executing folderId is 0 possibility.
```
DLFolder dlFolder = _dlFolderLocalService.getDLFolder(folderId);

		dlFolder.setName(UploadImageUtil.TEMP_IMAGE_FOLDER_NAME);

		_dlFolderLocalService.updateDLFolder(dlFolder);
```
2. Directly use the sql will get unexpected result. For example,  the user created one folder "java.lang.Class" in the UI. The fix will change it and this is not expected behavior. Please try to get specific dlFolder and update it. For example, I know we want to update the dlfolder which was from user profile. Please try to get the folder by using user profile info to get the folder.

**Comment by slnn** [comment]
Thank @yuhai !

Will update it according to the comment and resend it later.

**Pull request closed without merging**